tracer/src/Datadog.Trace/Iast/Aspects/System.Runtime/DefaultInterpolatedStringHandlerAspect.cs

```diff
@@ -0,0 +1,155 @@
+// <copyright file="DefaultInterpolatedStringHandlerAspect.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Iast.Propagation;
+using InlineIL;
+using static InlineIL.IL.Emit;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.Aspects.System.Runtime;
+
+/// <summary> DefaultInterpolatedString class aspect </summary>
+[AspectClass("System.Runtime")]
+[global::System.ComponentModel.Browsable(false)]
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public class DefaultInterpolatedStringHandlerAspect
+{
+    /// <summary>
+    /// System.Runtime DefaultInterpolatedStringHandler.AppendFormatted(String) aspect
+    /// </summary>
+    /// <param name="target"> the ref DefaultInterpolatedStringHandler </param>
+    /// <param name="value"> the string value </param>
+    [AspectMethodReplace("System.Runtime.CompilerServices.DefaultInterpolatedStringHandler::AppendFormatted(System.String)")]
+    public static void AppendFormatted1(ref DefaultInterpolatedStringHandler target, string value)
```

Comment:
You might also consider to add these overloads:

AppendFormatted(Object, Int32, String)	
AppendFormatted<T>(T, String)	
AppendFormatted<T>(T, Int32, String)

Actually, you could add unit tests in IastInstrumentationUnitTests.TestMethodsAspectCover and TestAllAspectsHaveACorrespondingMethod to make sure that you cover every method that you want to cover for every framework and that you don't have useless aspects for some frameworks

Reply:
We should cover relevant `string` related overloads, those used by the compiler when generating code for interpolated strings

Reply:
But if a string is declared as an object, we could potentially miss it. For instance, this test would fail:

    [Fact]
    public void GivenImplicitInterpolatedString_WhenAddingTaintedValuesComplex_GetString_Vulnerable()
    {
        var interpolatedString = $"""
                                  Hello "{((object)TaintedValue)}".
                                  .
                                  """;
        AssertTainted(interpolatedString);
    }

Reply:
Also, adding these unit test could detect if we are not covering a new string method in the DefaultInterpolatedStringHandler class that could be included in future versions of .Net

Reply:
😄 
- I applied in new commits the supports of this test case
- I also added the tests in the `IastInstrumentationUnitTests` class for the aspects checks

Reply:
Nice! Thank you!